tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -862,6 +862,18 @@ void SpirvEmitter::HandleTranslationUnit(ASTContext &context) {
                                 SourceLocation());
   }
 
+  // For Vulkan 1.2 and later, add SignedZeroInfNanPreserve when -Gis is
```

Comment:
For Vulkan < 1.2 the extension `SPV_KHR_float_controls` can be used to emit the `SignedZeroInfNanPreserve`. So should the execution mode always be emitted, and the extension presence depend on the env?